packages/shared/src/components/modals/BasePostModal.tsx

```diff
@@ -30,42 +33,49 @@ function BasePostModal({
   postPosition,
   onPreviousPost,
   onNextPost,
+  post,
   ...props
 }: BasePostModalProps): ReactElement {
+  const { usePostReferrer } = usePostReferrerContext();
+
+  usePostReferrer({ post });
+
   return (
-    <Modal
-      size={Modal.Size.XLarge}
-      kind={Modal.Kind.FlexibleTop}
-      portalClassName={styles.postModal}
-      id="post-modal"
-      {...props}
-      overlayClassName="post-modal-overlay bg-overlay-quaternary-onion"
-      className={classNames(
-        className,
-        'laptop: mx-auto !bg-background-default focus:outline-none tablet:h-full laptop:h-auto laptop:overflow-hidden',
-      )}
-    >
-      {isLoading ? (
-        <PostLoadingSkeleton
-          hasNavigation
-          type={postType}
-          className={loadingClassName}
-        />
-      ) : (
-        <>
-          <PostNavigation
-            className={{
-              container: 'px-4',
-            }}
-            postPosition={postPosition}
-            onPreviousPost={onPreviousPost}
-            onNextPost={onNextPost}
-            onClose={props?.onRequestClose}
+    <ActivePostContextProvider post={post}>
```

Comment:
Didn't @sshanzel  just introduce something like this context for post boost?
Want to avoid having many of the same'ish context

Reply:
@sshanzel I'm also blind, what was this called again?